### PR TITLE
Variable fonts

### PR DESCRIPTION
**Pull request by rlafuente**

### CODE REVIEW

**Comment by stuaxo** [comment]
Does anything else use the `var_` prefix ?

I'd be interested to see how it compares to passing in the dict of variants/axis directly.

```
font(..., axis={"wght": 400})
# or
font(..., variables={"wght": 400})
# or whatever name 
```

We probably want a way to query a font (or all fonts?) for available axis too ?

We should def have some examples with sliders for the font variables and see what the API is like in use.

**Comment by stuaxo** [comment]
Thanks for going through the bikeshedding on this !

File: shoebot/grammar/nodebox.py [hunk 1]
```diff
@@ -624,16 +622,36 @@ def background(self, *args):
...
+        :param vars: font variant values, as a dict of axis/value pairs (variable fonts only)
+        :param var_XXXX: set variant value (variable fonts only)
 
         :return: current current fontpath (if fontpath param not set)
         Accepts TrueType and OpenType files. Depends on FreeType being
         installed."""
         if fontpath is not None:
+            # do we have variants set?
+            if not vars:
+                variants = {}
```
Comment (stuaxo) [inline shoebot/grammar/nodebox.py:639]: Consider making variants a list of strings, like `["wdth=50", "wght=600"]` then the code below will get simpler...

The first loop can then be a list comprehension:

```python
variants = [f"{arg.replace('var_', '')={value}}" for arg, value in kwargs.items() if arg.startswith("var_")
```
# ^ Black does something interesting to the format of this..

Then the logic at the end can be something along the lines of 

```python
fontpath += " @" + variants.join(",")
```

Comment (rlafuente) [reply shoebot/grammar/nodebox.py]: great suggestion, pushed in d5f4b15056c93fc2de5906a37bbe3a49c0dda030

**Pull request merged**